modules/apps/digital-signature/digital-signature-web/test/js/components/document-previewer/DocumentPreviewerWrapper.js

```diff
@@ -19,7 +19,9 @@ import DocumentPreviewerWrapper from '../../../../src/main/resources/META-INF/re
 
 jest.mock('frontend-js-web', () => ({
 	...jest.requireActual('frontend-js-web'),
-	sub: jest.fn((string, arg) => string.replace(/x/g, arg)),
+	sub: jest.fn((key, ...args) =>
+		args.reduce((key, arg) => key.replace('x', arg), key)
+	),
```

Comment:
Why do we need to mock the implementation at all for sub? If we don't mock it does it cause a break?

Reply:
This test would pass without a mocked `sub`, but with updated snapshots.

- `document-1-of-1` would be `document-x-of-x`
- `the-document-Document-does-not-have-a-preview` would be `the-document-x-does-not-have-a-preview`
- etc.

Some tests would require way more adjustments if we were to altogether remove `sub` mocks.

The reason why all of a sudden we have some of these new mocks is because `Liferay.Util.sub` was mocked globally through our frontend-projects code. We could work on adding a sub mock there, I guess, so that it doesn't need to be defined in a lot of test files.

Reply:
All our tests would have `x` instead of the expected value wherever we used `sub`. Another example is this test with a formatted date if you remember from the other day - [BasicInformation.js](https://github.com/liferay/liferay-portal/blob/master/modules/dxp/apps/analytics/analytics-reports-web/test/js/components/BasicInformation.js#L66), this file is not a part of this PR, but the next one. If we don't mock `sub`, the text that `getByText` is looking for is `published-on-x`, so this test would need to be changed. 
How do you imagine we handle these kinds of cases?

Reply:
The reason I think the mock for `sub` is unhelpful here is because we are mocking a function that actually exists and works. Previously we had mocked `Liferay.Util.sub` because we didn't have a way to accurately test our implementation of that since we didn't have access to the global namespace of Liferay, so to handle that properly we added mocks for it. 

So for existing tests, it should be safe to update the snapshots from `document-1-of-1` to `document-x-of-x`. We don't really care that much whether it shows 1 or x because we are just making sure the snapshots are consistent and they reflect the actual implementation of `sub`.

For the case of [BasicInformation.js](https://github.com/liferay/liferay-portal/blob/master/modules/dxp/apps/analytics/analytics-reports-web/test/js/components/BasicInformation.js#L66), this is a case where we can add a single mock to this one test because it is dependent on the `x` value being properly replaced.

Another option would be to provide a global mock for `sub`. Currently `sub` is dependent on the transformation of language keys from `x-items` to `{0} items`. Since we don't have that in unit tests, we can add a global mock for `sub` to make things a bit more accurate. To add a global mock, I believe all you need to do is add a `__mocks__` folder and add a `sub.js` file at the same level of the current `sub.js` file

Reply:
Alright, sounds good. I'll just remove any changes here relating to `sub` and leave them for a different PR then, and squash commits here.

I wasn't sure where this conversation was heading last week (sounded like you were referring to this specific test), so thanks for clearing it up.